File: app/src/main/java/io/mgba/UI/Views/GameInformationView.java
package io.mgba.UI.Views;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.BottomSheetDialogFragment;
import android.support.design.widget.CoordinatorLayout;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import io.mgba.Constants;
import io.mgba.Data.Database.Game;
import io.mgba.R;
import io.mgba.UI.Activities.EmulationActivity;
import io.mgba.Utils.GlideUtils;
import io.mgba.Utils.GlideUtils.Colors;

public class GameInformationView extends BottomSheetDialogFragment implements View.OnClickListener {

    public static GameInformationView newInstance(Game game) {
        Bundle args = new Bundle();
        args.putParcelable(Constants.ARG_PLAY_GAME, game);

        GameInformationView frag = new GameInformationView();
        frag.setArguments(args);
        return frag;
    }

    @BindView(R.id.sheet_container) CoordinatorLayout main;
    @BindView(R.id.bottomsheet_header) RelativeLayout header;
    @BindView(R.id.gameTitle) TextView title;
    @BindView(R.id.gameDescription) TextView description;
    @BindView(R.id.cover) ImageView cover;
    @BindView(R.id.fab) FloatingActionButton playButton;
    @BindView(R.id.savestate_title) TextView savesHeader;
    @BindView(R.id.savestate_recyclerview) RecyclerView recyclerView;
    @BindView(R.id.no_content_image) ImageView noContentImage;
    @BindView(R.id.no_savestate_message) TextView noSavesMessage;
    @BindView(R.id.no_savestates_container) RelativeLayout noSavesContainer;

    private Game game;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.library_sheet_view, container, false);
        ButterKnife.bind(this, view);

        final Bundle arguments = getArguments();

        if(arguments != null)
            this.game = arguments.getParcelable(Constants.ARG_PLAY_GAME);

        prepareView();

        return view;
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        outState.putParcelable(Constants.ARG_SHEET_CONTENT, game);
    }


    private void prepareView() {
        title.setText(game.getName());
        description.setText(game.getDescription());

        if(cover != null && game.getCoverURL() != null) {
            GlideUtils.init(this, game.getCoverURL())
                      .setPlaceholders(R.drawable.placeholder, R.drawable.error)
                      .colorView(Colors.VIBRANT, Colors.DARK_MUTED, playButton, savesHeader)
                      .colorView(Colors.LIGHT_MUTED, Colors.LIGHT_VIBRANT, header, noSavesMessage, noContentImage)
                      .colorView(Colors.LIGHT_VIBRANT, true, title)
                      .colorView(Colors.LIGHT_VIBRANT, false, description)
                      .build(cover);
        }
    }

    @Override
    @OnClick(R.id.fab)
    public void onClick(View v) {
        Intent it = new Intent(getContext(), EmulationActivity.class);
        it.putExtra(Constants.ARG_PLAY_GAME, game);
        getContext().startActivity(it);
    }
}

File: app/src/main/java/io/mgba/ui/views/GameInformationView.kt
package io.mgba.ui.views

import android.content.Intent
import android.graphics.drawable.GradientDrawable
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import io.mgba.R
import io.mgba.data.local.model.Game
import io.mgba.emulation.EmulationActivity
import io.mgba.utilities.io.GlideUtils
import io.mgba.utilities.io.GlideUtils.Colors
import kotlinx.android.synthetic.main.game_information_view.*
import java.lang.RuntimeException

class GameInformationView : BottomSheetDialogFragment(), View.OnClickListener {

    private lateinit var game: Game

    override fun getTheme(): Int {
        return R.style.BottomSheetDialogTheme
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.game_information_view, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        if(arguments == null) throw RuntimeException("The bottomsheet could retrive the game_item!")

        arguments?.let {
            //this.game = it.getParcelable(ARG_PLAY_GAME)!!
            prepareView()
        }
    }


    private fun prepareView() {
        game_title_bs.text = game.name
        game_desc_bs.text = game.description

        if (game_cover_bs != null && game.cover != null) {
            GlideUtils.init(this, game.cover!!)
                    .setPlaceholders(R.drawable.placeholder, R.drawable.error)
                    .colorView(Colors.VIBRANT, Colors.DARK_MUTED, listOf(game_play_bs, game_savetitle_bs))
                    .colorView(Colors.LIGHT_MUTED, Colors.LIGHT_VIBRANT, listOf(game_text_content_bs, game_image_content_bs))
                    .colorViewWithCustomBackground(Colors.LIGHT_MUTED, Colors.LIGHT_VIBRANT, listOf(game_header_bs))
                    .colorView(Colors.LIGHT_VIBRANT, true, listOf(game_title_bs))
                    .colorView(Colors.LIGHT_VIBRANT, false, listOf(game_desc_bs))
                    .build(game_cover_bs)
        } else {
            //set header color back to normal
            val background = game_header_bs.background

            if (background is GradientDrawable) {
                background.setColor(ContextCompat.getColor(context!!, R.color.colorPrimary))

            }
        }
    }

    override fun onClick(v: View) {
        val it = Intent(context, EmulationActivity::class.java)
        //it.putExtra(ARG_PLAY_GAME, game)
        requireContext().startActivity(it)
    }

    companion object {

        fun show(game: Game): GameInformationView {
            val args = Bundle()
            //args.putParcelable(Constants.ARG_PLAY_GAME, game)

            val frag = GameInformationView()
            frag.arguments = args
            return frag
        }
    }
}
